src/style-manager.js

```diff
@@ -1,17 +1,28 @@
+import { Map, List } from 'immutable';
+
 export default class StyleManager {
     constructor() {
-        this.rules = {};
+        this._rules = Map();
+    }
+
+    get rules() {
+        return this._rules.toJS();
+    }
+
+    set rules(rules) {
+        this._rules = rules;
     }
 
     change(namespace) {
         return Object.assign(this, {
             when: condition => {
-                if (!this.rules[namespace]) this.rules[namespace] = [];
+                if (!this._rules.get(namespace)) this._rules = this._rules.set(namespace, List([]));
```

Comment:
`List` constructor should by default generate an empty list - so no need to specify an empty list object as its initial value.